Automattic/wp-calypso
# Persist sidebar state from Calypso to User calypso_preferences.

#### Changes proposed in this Pull Request

* Save sidebar state to calypso_preferences.
* Read sidebar state from calypso_preferences.

#### Testing instructions



* Click "Collapse menu"
* Make sure the sidebar collapses
* Inspect the indexed DB and check that `"redux-state-[USER_ID]:preferences"` object has a key `sidebarCollapsed: true`
* Click "Collapse menu" again
* Reload the Indexed DB values and inspect that `"redux-state-[USER_ID]:preferences"` object has a key `sidebarCollapsed: false`
* Click "Collapse menu" again
* Reload the page, make sure that the sidebar remains collapsed
* Click on Switch Site or Site Card or Nudges, make sure the sidebar expands
* Reload the Indexed DB values and inspect that `"redux-state-[USER_ID]:preferences"` object has a key `sidebarCollapsed: false`

If reviewed in conjunction with https://github.com/Automattic/jetpack/pull/19520, you may also check that SIMPLE sites and Calypso are in sync (There is PR pending for propagating the preference to Atomic sites).



Fixes #46645
